MSAL/src/native_auth/controllers/sign_up/MSALNativeAuthSignUpControlling.swift

```diff
@@ -26,14 +26,13 @@
 
 protocol MSALNativeAuthSignUpControlling: AnyObject {
 
-    typealias SignUpStartPasswordControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpPasswordStartResult>
-    typealias SignUpStartCodeControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpStartResult>
+    typealias SignUpStartControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpStartResult>
     typealias SignUpSubmitCodeControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpVerifyCodeResult>
     typealias SignUpSubmitPasswordControllerResponse = MSALNativeAuthControllerTelemetryWrapper<SignUpPasswordRequiredResult>
 
-    func signUpStartPassword(parameters: MSALNativeAuthSignUpStartRequestProviderParameters) async -> SignUpStartPasswordControllerResponse
+    func signUpStartPassword(parameters: MSALNativeAuthSignUpStartRequestProviderParameters) async -> SignUpStartControllerResponse
```

Comment:
Now in the controller we have two different methods that return the same response:

`func signUpStartPassword() -> SignUpStartControllerResponse`
`func signUpStartCode() -> SignUpStartControllerResponse`

I think we could simplify the controller interface by using one single method (like in the public interface), and let it decide internally whether use `private signUpStartPassword()` or `private signUpStartCode()`.

Reply:
We could remove a lot of almost duplicated code, in fact. But a had to draw a line somewhere, since this changes were quite extensive. With this interface modification, we are already avoiding breaking changes in the near future, which is the main objective here. We could add this extra refactoring to a new PBI.